### PR TITLE
DDF-1993 - updated RestEndpoint to URL decode sourceId & metacardId

### PR DESCRIPTION
**Pull request by kcwire**
#### What does this PR do?
Fixes bug where you cannot execute Actions when the site name has a space in it.
#### Who is reviewing it?
@jlcsmith @pklinef 
#### How should this be tested?
Update the site name to have a space, ingest some data, verify the actions work from the search ui.
#### What are the relevant tickets?
DDF-1993
#### Checklist:
N/A - [ ] Documentation Updated
- [ x ] Update / Add Unit Tests
N/A - [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java [hunk 1]
```diff
@@ -417,7 +420,7 @@ public Response getDocument(@Context UriInfo uriInfo, @Context HttpServletReques
      */
     @GET
     @Path("/sources/{sourceid}/{id}")
-    public Response getDocument(@PathParam("sourceid") String sourceid, @PathParam("id") String id,
+    public Response getDocument(@PathParam("sourceid") String encodedSourceId, @PathParam("id") String encodedId,
```
Comment (stustison) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java]: Anytime you're assuming that a JAXRS parameter is encoded, you should also be using `@Encoded` to ensure that it does not get decoded, which will then cause the URLDecoder to give you incorrect results.

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java [hunk 2]
```diff
@@ -430,8 +433,8 @@ public Response getDocument(@PathParam("sourceid") String sourceid, @PathParam("
         URI absolutePath = uriInfo.getAbsolutePath();
         MultivaluedMap<String, String> map = uriInfo.getQueryParameters();
 
-        if (id != null) {
-            LOGGER.debug("Got id: {}", id);
+        if (encodedId != null) {
+            LOGGER.debug("Got id: {}", encodedId);
```
Comment (jlcsmith) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java]: Nit- all these separate debug statements should be merged together and cleaned up.

File: catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java [hunk 3]
```diff
@@ -452,7 +457,8 @@ public Response getDocument(@PathParam("sourceid") String sourceid, @PathParam("
                         .text(id);
 
                 Collection<String> sources = null;
-                if (sourceid != null) {
+                if (encodedSourceId != null) {
+                    String sourceid = URLDecoder.decode(encodedSourceId, CharEncoding.UTF_8);
                     sources = new ArrayList<String>();
                     sources.add(sourceid);
```
Comment (jlcsmith) [inline catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java]: `sources.add(URLDecoder.decode(encodedSourceId, CharEncoding.UTF_8));`

**Comment by jlcsmith** [comment]
Need to rebase with master please.

**Pull request merged**